equinix/resource_network_device.go

```diff
@@ -298,7 +302,7 @@ func createNetworkDeviceSchema() map[string]*schema.Schema {
 		neDeviceSchemaNames["IsBYOL"]: {
 			Type:        schema.TypeBool,
 			Optional:    true,
-			Default:     false,
+			Default:     true,
```

Comment:
Are customers currently using this resource?  I'm a bit concerned about changing this default, especially since this attribute is `ForceNew`, so if a customer fails to notice the change, they could accidentally delete something.

Reply:
Yes, they are using the resource to create the device. Since we will publish a new version for terraform provider with the new change, the customer should know this change of the default value. Like I mentioned in the description of the PR, for all the current supporting vendors, we no longer offer the Equinix managed option, same for this license mode, BYOL will be the major one (Only limited device types would have subscription option). It would be good to change the default value to be self configured and BYOL. If they specify the field in their script, they should not see any changes. If not, terraform would at least prompt the changes to warn the customer. 

If a customer uses the previous version of terraform provider to provision the device, are they allowed to upgrade the provider version to manage the existing deployment? This is the only situation that will cause the problem you mentioned above. Wish to hear more advice for the scenario.

Reply:
It is possible for a customer to automatically pull in the new provider version for an existing deployment; it depends on how they have set up their [provider version constraints](https://developer.hashicorp.com/terraform/language/providers/requirements#version-constraints).  We may be able to reduce the likelihood of a customer automatically pulling in this default value change by bumping the major version instead of the minor, but since the terraform provider encompasses multiple resources across multiple services, we may not want to do a major version bump in order to change a single default value.